spec/index.bs

```diff
@@ -884,20 +884,90 @@ The [=subresources=] of textures included in the views provided to
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depthOrArrayLayers=].
+
+            : {{GPUTextureDimension/"3d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to max(1, |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] &Gt; |mipLevel|).
+
+                Set |extent|.{{GPUExtent3DDict/height}} to max(1, |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] &Gt; |mipLevel|).
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to max(1, |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depthOrArrayLayers=] &Gt; |mipLevel|).
+        </dl>
+    1. Return |extent|.
+</div>
+
+The <dfn dfn>physical miplevel-specific texture extent</dfn> of a [=texture=] is the size of the
+[=texture=] in texels at a specific miplevel that includes the possible extra padding
+to form complete [=texel blocks=] in the [=texture=]. It is calculated by this procedure:
+
+<div algorithm>
+    <dfn abstract-op>Physical miplevel-specific texture extent</dfn>(descriptor, mipLevel)
+
+    **Arguments:**
+        - {{GPUTextureDescriptor}} |descriptor|
+        - {{GPUSize32}} |mipLevel|
+
+    **Returns:** {{GPUExtent3DDict}}
+
+    1. Let |extent| be a new {{GPUExtent3DDict}} object.
+    1. Let |logicalExtent| be [=logical miplevel-specific texture extent=](|descriptor|, |mipLevel|).
+    1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is:
+        <dl class="switch">
+            : {{GPUTextureDimension/"1d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to 1.
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |logicalExtent|.[=Extent3D/depthOrArrayLayers=].
+
+            : {{GPUTextureDimension/"2d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to |logicalExtent|.[=Extent3D/height=] rounded up to the nearest multiple of |descriptor|'s [=texel block height=].
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |logicalExtent|.[=Extent3D/depthOrArrayLayers=].
+
+            : {{GPUTextureDimension/"3d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to |logicalExtent|.[=Extent3D/height=] rounded up to the nearest multiple of |descriptor|'s [=texel block height=].
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |logicalExtent|.[=Extent3D/depthOrArrayLayers=].
```

Comment:
We don't have 3d textures with block formats (yet), but we may end up with a `texel block depth` at some point:
https://en.wikipedia.org/wiki/Adaptive_scalable_texture_compression#3D_block_footprints_and_bit_rates

So maybe we should have something here like `[=Assert=] that |descriptor|.{{GPUTextureDescriptor/format}} is not a block format`

Reply:
I'm a little uncomfortable putting asserts into a web spec, because they mean nothing but they're also kind of FUD... The intended audience for this claim is neither implementors nor authors.

Reply:
They're established here: https://infra.spec.whatwg.org/#assertions
and I think they are useful to implementers because they tell you invariants that are supposed to hold true (and could be debug-asserted in implementation code)

Reply:
In this case such a debug-assert in implementation code would be useful because when you try to add and test 3d compressed textures you'll discover that this hasn't been fully implemented (doesn't handle block depths if those turn out to exist).